crates/solidity/outputs/npm/tests/src/cst-cursor.ts

```diff
@@ -0,0 +1,264 @@
+import { Language } from "@nomicfoundation/slang/language";
+import { NodeType, RuleNode, TokenNode } from "@nomicfoundation/slang/cst";
+import { ProductionKind, TokenKind } from "@nomicfoundation/slang/kinds";
+import { Cursor } from "@nomicfoundation/slang/cursor";
+
+test("use cursor", () => {
+  const source = "int256 constant z = 1 + 2;";
+  const language = new Language("0.8.1");
+
+  const { parseTree } = language.parse(ProductionKind.SourceUnit, source);
+  const cursor: Cursor = parseTree.cursor;
+
+  expect(cursor.node.type).toBe(NodeType.Rule);
```

Comment:
nit, I suggest creating an array of the expectations and iterate on them instead. IMO, It is much harder to understand and/or update the contents otherwise.